service-base/src/test/java/org/eclipse/hono/service/metric/MicrometerBasedMetricsTest.java

```diff
@@ -218,4 +219,99 @@ public void testPayloadSizeForCommandMessages(final MeterRegistry registry) {
                 registry.find(MicrometerBasedMetrics.METER_COMMANDS_PAYLOAD).summary().totalAmount());
     }
 
+    /**
+     * Verifies that the metrics for a tenant are removed when all their connections are closed.
+     *
+     * @param registry The registry that the tests should be run against.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testAuthenticatedConnections(final MeterRegistry registry) {
+
+        final MicrometerBasedMetrics metrics = new MicrometerBasedMetrics(registry);
+
+        final String tenant1 = "tenant1";
+        final String tenant2 = "tenant2";
+
+        // GIVEN a metrics instance that has recorded one telemetry message and one event per tenant
+        metrics.incrementConnections(tenant1);
+        metrics.incrementConnections(tenant1);
+        metrics.incrementConnections(tenant2);
+        metrics.incrementConnections(tenant2);
+
+        final Search search = registry.find(MicrometerBasedMetrics.METER_CONNECTIONS_AUTHENTICATED);
+        assertEquals(2, search.meters().size());
+        assertEquals(4, metrics.getNumberOfConnections());
+
+        // WHEN closing all connections for one tenant
+        metrics.decrementConnections(tenant1);
+        assertEquals(2, search.meters().size());
+        metrics.decrementConnections(tenant1);
+
+        // THEN the metric for this tenant is removed
+        assertEquals(1, search.meters().size());
```

Comment:
IMHO you should make sure that the correct metric has been removed, i.e. the one for tenant1 ...